tests/Platforms/SQLitePlatformTest.php

```diff
@@ -635,4 +638,50 @@ public function testGetCreateTableSQLWithColumnCollation(): void
             $this->platform->getCreateTableSQL($table),
         );
     }
+
+    #[DataProvider('sqliteTemporaryProvider')]
+    public function testGenerateTemporaryTable(
+        bool $temporary,
+        string $expectedSQL,
+    ): void {
+        $table = new Table('mytable');
+        $table->addOption('temporary', $temporary);
+
+        $table->addColumn('foo', Types::STRING, ['length' => 255]);
+
+        self::assertEquals(
+            [$expectedSQL],
+            $this->platform->getCreateTableSQL($table),
+        );
+    }
+
+    public static function sqliteTemporaryProvider(): Generator
+    {
+        yield 'temporary' => [true, 'CREATE TEMPORARY TABLE mytable (foo VARCHAR(255) NOT NULL)'];
+        yield 'non temporary' => [false, 'CREATE TABLE mytable (foo VARCHAR(255) NOT NULL)'];
+    }
+
+    #[DataProvider('sqliteInvalidTemporaryProvider')]
+    public function testInvalidTemporaryTableOptions(
+        string $table,
+        mixed $temporary,
+        string $expectedException,
+        string $expectedMessage,
+    ): void {
+        $this->expectException($expectedException);
+        $this->expectExceptionMessage($expectedMessage);
+
+        $table = new Table($table);
+        $table->addOption('temporary', $temporary);
+
+        $table->addColumn('foo', Types::STRING, ['length' => 255]);
+
+        $this->platform->getCreateTableSQL($table);
+    }
+
+    public static function sqliteInvalidTemporaryProvider(): Generator
+    {
+        yield 'invalid temporary specification' =>
+        ['mytable', 'invalid', InvalidArgumentException::class, 'invalid temporary specification for table mytable'];
+    }
```

Comment:
Please don't create data provides with only a single set of data.

Reply:
Noted and removed